[nictjh] iP

nictjh opened the pull request:
### **Solace**
> *"A clear mind leads to a peaceful life"* - TalkCock

Solace helps you **declutter your mind and stay on top of your tasks effortlessly.** With Solacem you can focus on what truly matters to you!

## **Why Solace?**
- ~~Simple~~ VERY SIMPLE and Intuitive
- Test-Based Interface

## Getting Started
1. Download it from [here](https://github.com/nictjh/ip).
2. Open the application
3. Add your tasks
4. Stay organized without the stress.

✨ **And yes, it’s completely FREE!** ✨  

## **Features**
- [x] Task Management - Keep track of everything in one place
- [x] Deadlines & Due dates
- [ ] Smart Reminders *(coming soon)*

## Usage
### 1. Adding a normal task
```plaintext
todo Buy groceries

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nictjh committed:
Changed to Solace

nictjh committed:
Add echo capabilities

nictjh committed:
Add list ability

nictjh committed:
Add marking ability

nictjh committed:
Add different tasks

nictjh committed:
Implement automated text testing

nictjh committed:
Add exception handling

nictjh committed:
Add deleting tasks ability

nictjh committed:
Add save and load ability

nictjh committed:
Merge branch with save and load functions

nictjh committed:
Add date and time recognition and find

nictjh committed:
Merge branch with date, time and find function

nictjh committed:
Add new classes and refactor

nictjh committed:
Merge branch with new oop

nictjh committed:
Edit white spaces and structure

nictjh committed:
Merge branch 'branch-A-Packages'
Merge branch with package structure

nictjh committed:
Modify slightly

nictjh committed:
Merge branch 'branch-testing'
Merge branch with the slight modifications

nictjh committed:
Merge branch 'add-gradle-support'
Merging branch to have gradle support

nictjh committed:
Sync gradle with project

nictjh committed:
Add checkstyle from addressbook

nictjh committed:
Merge branch with updated gradle info

nictjh committed:
Add JUnit test for two methods

nictjh committed:
Add the two classes for testing

nictjh committed:
Merge branch 'branch-A-JUnit'
Merge branch with some JUnit tests

nictjh committed:
Modify code to build jar file

nictjh committed:
Merge branch with the build details

nictjh committed:
Add JavaDocs for methods and class

nictjh committed:
Merge branch 'branch-A-JavaDoc'
Merge branch with javaDocs written

nictjh committed:
Improve on checkstyle

nictjh committed:
Merge branch 'branch-A-CodingStandard' with conflict resolution

nictjh committed:
Fix missing import due to merge resolution

nictjh committed:
Add find ability and JavaDocs for JUnit

nictjh committed:
Merge branch with find function and more JavaDocs

SociallyIneptWeeb commented on src/main/java/app/Solace.java:
```diff
@@ -0,0 +1,87 @@
+public class Solace {
+
+    private static boolean isAlive;
+    private final TaskList TASKLIST;
+    private final String FILE_PATH = "bin/storage";
+    private static Ui UI;
+    private final Storage STORAGE;
```
Not too sure if the tasklist, ui and storage variables should be capitalized instead of camelCase. Is it because you are treating these objects as constants?

nictjh replied:
Thanks for pointing it out man, I might have misinterpreted the standards a bit!

SociallyIneptWeeb commented on src/main/java/commands/CommandParser.java:
```diff
@@ -0,0 +1,104 @@
+                return new UnmarkCommand(Integer.parseInt(inputArray[1]));
+            case "todo":
+                if (inputArray.length < 2) {
+                    throw new MissingToDoDescException();
+                }
+                String toDoDesc = extractDescription(inputArray);
+                return new CreateToDoCommand(toDoDesc);
+            case "deadline":
+                if (inputArray.length < 2 || !input.contains("/by")) {
+                    throw new MissingDeadlineDetailsException();
+                }
+                String deadlineDesc = extractDescription(inputArray);
+                String dDesc = deadlineDesc.substring(0, deadlineDesc.indexOf("/by"));
+                String by = deadlineDesc.substring(deadlineDesc.indexOf("/by") + 4);
+                return new CreateDeadlineCommand(dDesc, by);
+            case "event":
+                if (inputArray.length < 2 || !input.contains("/from") || !input.contains("/to")) {
+                    throw new MissingEventDetailsException();
+                }
+                String eventDesc = extractDescription(inputArray);
+                String eDesc = eventDesc.substring(0, eventDesc.indexOf("/from"));
+                String from = eventDesc.substring(eventDesc.indexOf("/from") + 6, eventDesc.indexOf("/to"));
+                String to = eventDesc.substring(eventDesc.indexOf("/to") + 4);
+                return new CreateEventCommand(eDesc, from, to);
+            case "delete":
+                validateArguments(inputArray, 2);
+                int deleteIndex = Integer.parseInt(inputArray[1]);
+                return new DeleteTaskCommand(deleteIndex);
+            case "findDate":
+                validateArguments(inputArray, 2);
+                String findString = inputArray[1];
+                return new FindDateCommand(findString);
+            case "find":
+                String desc = extractDescription(inputArray);
+                return new FindCommand(desc);
+            case "save":
+                return new SaveCommand(FILE_PATH);
+            default:
+                return new UnknownCommand(); // throws error need to be caught in app.Solace
+        }
```
There is no indentation for case clauses for switch statements, you may want to configure your IDE to follow this style instead.

nictjh replied:
Will change it immediately man, thanks for the suggestion!

SociallyIneptWeeb commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,43 @@
+        this.isDone = false;
+    }
+
+    public boolean getStatus() {
+        return isDone;
+    }
```
Boolean variables/methods should be named to sound like booleans, so maybe rename it to something like isDone?

nictjh replied:
thanks for catching my errors haha

SociallyIneptWeeb commented on src/main/java/taskList/TaskList.java:
```diff
@@ -0,0 +1,235 @@
+        return defMsg + task.toString() + "\n";
+    }
+
+    /**
+     * Removes a task from the list
+     *
+     * @param num The index of the task to be removed, this is 1-indexed
+     * @return A String message indicating the task has been removed
+     * @throws InvalidTaskNumberException InvalidTaskNumberException If the task number is invalid
+     * @throws EmptyTaskListException EmptyTaskListException If the task list is empty
+     */
```
For javadoc comments, you may want to add punctuation at the end of each parameter description.

nictjh replied:
for sure man

SociallyIneptWeeb commented on src/test/java/commands/CreateDeadlineCommandTest.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Test
+    void testCreateDeadlineCommand() {
```
You may want to follow the convention for test method names: `featureUnderTest_testScenario_expectedBehavior()`.

nictjh replied:
will follow this !

SociallyIneptWeeb reviewed (commented):
Overall, you followed most of the coding standards! Just a few nits here and there.

nictjh committed:
Check and correct to code standards

nictjh committed:
Merge branch 'branch-A-Checkstyle'
Merge branch with checkstyle

nictjh committed:
Integrate basic Gui

nictjh committed:
Integrate MainWindow fxml

nictjh committed:
Merge branch with GUI integration

nictjh committed:
Add styling and keycode function

nictjh committed:
Merge branch with new styling

nictjh committed:
Add java varargs into mark command

nictjh committed:
Merge branch 'branch-A-Varargs'
Merge branch with new MarkCommand

nictjh committed:
Add assertions to validate assumptions made

Currently the codebase lacks assertions to document critical assumptions,
which may result in errors like IndexOutofBoundsException etc w/o early
detection

This needs to change because the absence of assertions can make
debugging harder and allows silent failures to propagate

Added assertions to validate method inuts, object states
in Storage, TaskList and CommandParser

This is done to improve debugging by catching
incorrect assumptions early. They are placed where
assumptions should always be true

nictjh committed:
Merge pull request #1 from nictjh/branch-A-Assertions

Add assertions to validate assumptions made

nictjh committed:
Improve naming conventions and documentations

Currently, some classes uses vague variable names which
do not clearly convey their purpose, making it hard
to understand their roles. Additionally, there were
missing JavaDocs as well.

This needs to change because adhering to coding
standards helps improves readability, and reduces
ambiguity.

Refactor variable names to better represent their
respective meanings and updated JavaDoc comments

This is done to align with the naming conventions
and best practices

nictjh committed:
Merge pull request #2 from nictjh/branch-A-CodeQuality

Improve naming conventions and documentations

nictjh committed:
Add sort to print tasks chronologically

nictjh committed:
Merge branch with sorting capability

nictjh committed:
Fix text truncation issue in DialogBox

DialogBox was using lavel to display msg, which led to text
being truncated when the message length exceeded the avail
width. This resulted in an inconsistent and frustrating ux
as longer messages were cut off rather than wraped to next
line.

TextFlow provides better control over text layout and
wrapping. Unlike label, it will automatically adjust
the text within the width, so messages are displayed
in full without truncation.

Let's replace Label with TextFlow in DialogBox to:
- Improve text wrapping and prevent truncation
- Allow dynamic resizing and better handling of long msg

The max width for TextFLow is set to 500px to maintain a
clean layout without strething too wide. So the bubbles
are still easy to read.

This approach was chosen because TextFlow offers more
flexibility and control compared to label.

nictjh committed:
Merge branch with better GUI

nictjh committed:
Add more personality for bot startup and end messages

nictjh committed:
Merge branch with more personality

nictjh committed:
Update README with AI tools

nictjh committed:
Merge branch with new README

nictjh committed:
Handle running in new environment

nictjh committed:
Merge branch 'branch-A-MoreErrorHandling'
Merge branch with saving ability in new env

nictjh committed:
Refactor code to ensure SLAP

nictjh committed:
Merge branch with more readable code

nictjh committed:
Update doc/README and add UI image

nictjh committed:
Merge branch with updated docs and image

nictjh committed:
Add untracked files

nictjh committed:
Merge branch 'branch-UntrackedFiles'
Merge branch with important untracked files
